### PR TITLE
Blobstore uploads

### LINKED ISSUE #26: Enable upload and storage of project files

### LINKED ISSUE #51: Enable project file downloads

### PR DESCRIPTION
**Pull request by gwynethsc**
Enable file upload at the project detail page using Blobstore with Java servlet backend. File names are listed and temporarily link directly to serving URLs to verify working uploads.

Enabling uploads requires running the backend server at step71-2020/project separately with `mvn appengine:run`

### CODE REVIEW

File: angular-app/src/app/left-menu/create-project-dialog.html
Comment (Google-JM) [inline angular-app/src/app/left-menu/create-project-dialog.html]: More descriptive class name?

File: angular-app/src/app/left-menu/left-menu.component.ts
Comment (Google-JM) [inline angular-app/src/app/left-menu/left-menu.component.ts]: This component should probably be declared in another file for organization purposes. create-dialog.component.ts

File: angular-app/src/app/project-detail/project-detail.component.ts
Comment (Google-JM) [inline angular-app/src/app/project-detail/project-detail.component.ts]: Is this test data needed any more if you can now upload blob with this pull?

**Review by Google-JM — CHANGES_REQUESTED** [review]
This is looking great.

From a high-level team perspective:
It seem like file upload is tightly coupled to the project creation page. From my understanding the workspace also uses this functionality. You may want to make the file upload into a component that can be easily added anywhere. You and Jon should talk about this. I also recommend you do not submit this until Jon reviews and approves to ensure you are on the same page on storage of media.

File: angular-app/src/app/project.service.ts
Comment (jeward414) [inline angular-app/src/app/project.service.ts:64]: Is the blobstoreUpload where the media files will go? Then the text doc will be saved as...?

File: angular-app/src/app/project-detail/project-detail.component.ts
Comment (gwynethsc) [reply angular-app/src/app/project-detail/project-detail.component.ts]: I thought I'd already deleted it . . . done now

File: angular-app/src/app/left-menu/create-project-dialog.html
Comment (gwynethsc) [reply angular-app/src/app/left-menu/create-project-dialog.html]: Removed altogether

File: angular-app/src/app/left-menu/left-menu.component.ts
Comment (gwynethsc) [reply angular-app/src/app/left-menu/left-menu.component.ts]: Done

File: angular-app/src/app/project.service.ts
Comment (gwynethsc) [reply angular-app/src/app/project.service.ts:64]: The text document can probably be saved directly into Firestore

**Comment by gwynethsc** [comment]
Note that chat has been removed from the application in this PR, although the component files are kept for potential later use.

File: project/src/main/java/com/google/ensemble/servlets/UploadServlet.java
Comment (sperzion) [inline project/src/main/java/com/google/ensemble/servlets/UploadServlet.java]: You can inline `firestoreOptions` since it's not used anywhere else, and it's likely not going to be used anywhere else.

File: project/src/main/java/com/google/ensemble/servlets/UploadServlet.java
Comment (sperzion) [inline project/src/main/java/com/google/ensemble/servlets/UploadServlet.java]: These strings probably should be constants.

File: project/src/main/java/com/google/ensemble/servlets/UploadServlet.java
Comment (sperzion) [inline project/src/main/java/com/google/ensemble/servlets/UploadServlet.java]: These too

File: project/src/main/java/com/google/ensemble/servlets/UploadServlet.java
Comment (sperzion) [inline project/src/main/java/com/google/ensemble/servlets/UploadServlet.java]: This is tying your back end to your front end. It's ok to say that your service only handles one file upload at a time. You might even consider throwing an exception if you get more than 1.

**Review by leah-mayo — APPROVED** [review]

File: project/src/main/java/com/google/ensemble/servlets/UploadServlet.java
Comment (gwynethsc) [reply project/src/main/java/com/google/ensemble/servlets/UploadServlet.java]: Done

File: project/src/main/java/com/google/ensemble/servlets/UploadServlet.java
Comment (gwynethsc) [reply project/src/main/java/com/google/ensemble/servlets/UploadServlet.java]: Done

File: project/src/main/java/com/google/ensemble/servlets/UploadServlet.java
Comment (gwynethsc) [reply project/src/main/java/com/google/ensemble/servlets/UploadServlet.java]: Changed so that the backend support multiple file uploads

File: project/src/main/java/com/google/ensemble/servlets/UploadServlet.java
Comment (gwynethsc) [reply project/src/main/java/com/google/ensemble/servlets/UploadServlet.java]: Done

**Review by Google-JM — APPROVED** [review]
LGTM after you resolve all your threads with Justin H.

**Pull request merged**
